RiotNSE/NotificationService.swift

```diff
@@ -0,0 +1,499 @@
+    var originalContent: UNMutableNotificationContent?
+    
+    var userAccount: MXKAccount?
+    var store: MXFileStore?
+    var showDecryptedContentInNotifications: Bool {
+        return RiotSettings.shared.showDecryptedContentInNotifications
+    }
+    
+    override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+        self.contentHandler = contentHandler
+        //  save this content as fallback content
+        originalContent = request.content.mutableCopy() as? UNMutableNotificationContent
+        
+        UNUserNotificationCenter.current().removeUnwantedNotifications()
+        
+        //  check if this is a Matrix notification
+        guard let content = originalContent else {
+            return
+        }
+        
+        let userInfo = content.userInfo
+        NSLog("[NotificationService] Payload came: \(userInfo)")
+        let roomId = userInfo["room_id"] as? String
+        let eventId = userInfo["event_id"] as? String
+
+        guard roomId != nil, eventId != nil else {
+            //  it's not a Matrix notification, do not change the content
+            NSLog("[NotificationService] Fallback case 7")
+            contentHandler(content)
+            return
+        }
+        
+        //  setup user account
+        setup()
+        
+        //  fetch the event first
+        fetchEvent()
+    }
+    
+    override func serviceExtensionTimeWillExpire() {
+        // Called just before the extension will be terminated by the system.
+        // Use this as an opportunity to deliver your "best attempt" at modified content, otherwise the original push payload will be used.
+        NSLog("[NotificationService] Fallback case 5")
+        fallbackToOriginalContent()
+    }
+    
+    func setup() {
+        let sdkOptions = MXSDKOptions.sharedInstance()
+        sdkOptions.applicationGroupIdentifier = "group.im.vector"
+        sdkOptions.disableIdenticonUseForUserAvatar = true
+        sdkOptions.enableCryptoWhenStartingMXSession = true
+        sdkOptions.backgroundModeHandler = MXUIKitBackgroundModeHandler()
+        Bundle.mxk_customizeLocalizedStringTableName("Vector")
+        
+        if isatty(STDERR_FILENO) == 0 {
+            MXLogger.setSubLogName("nse")
+            MXLogger.redirectNSLog(toFiles: true)
+        }
+        
+        userAccount = MXKAccountManager.shared()?.activeAccounts.first
+        
+        if let userAccount = userAccount {
+            store = MXFileStore(credentials: userAccount.mxCredentials)
+            
+            if userAccount.mxSession == nil {
+                userAccount.openSession(with: store!)
+            }
+        }
+    }
+    
+    func fetchEvent() {
+        guard let content = originalContent, let userAccount = self.userAccount else {
+            //  there is something wrong, do not change the content
+            NSLog("[NotificationService] Fallback case 4")
```

Comment:
Could we have more detail in the log for future debugging? 
Like:
```suggestion
            NSLog("[NotificationService] fetchEvent: Fallback case 4. There is something wrong, do not change the content")
```

This worths for all `Fallback case X` :)

I do not understand the order for X :/

Reply:
You're totally right. It was easy for debugging on development. Changing them all.